jetty-ee11/jetty-ee11-webapp/src/main/java/org/eclipse/jetty/ee11/webapp/WebAppClassLoader.java

```diff
@@ -575,6 +580,41 @@ protected Class<?> foundClass(final String name, URL url) throws ClassNotFoundEx
         }
     }
 
+    private void definePackageIfNecessary(String className, URL url) throws IOException
+    {
+        int lastDotIndex = className.lastIndexOf('.');
+        if (lastDotIndex < 0)
+            return;
+        String packageName = className.substring(0, lastDotIndex);
+        if (getDefinedPackage(packageName) == null)
+        {
+            try
+            {
+                String externalForm = url.toExternalForm();
+                if (externalForm.startsWith("jar:file:") && externalForm.contains("!/"))
```

Comment:
Is there any chance this could be coming from an `extraClasspath` that is a directory, ie a `file:` url?

Reply:
@janbartel Sure it can... in which case it falls down to the package below.  A package is still defined, but without any data from a manifest, which we can only get if it is from a jar.

Reply:
I'll add a test for this....

Reply:
@janbartel done